guide/api-design-guidelines.md

```diff
@@ -0,0 +1,170 @@
+
+`SecretReference` and `SecretConfig` are entities prescribed to ensure sensitive information is handled securely and consistently across different components and resources in Radius. The following are their definitions along with usage examples:
+
+### SecretReference Model
+
+<a href="#secret-model" name="secret-model">:white_check_mark:</a> **DO** follow this structure when adding support for secrets to resources or components in Radius.
+
+```
+
+@doc("This specifies a reference to a secret. Secrets are encrypted, often have fine-grained access control, auditing and are recommended to be used to hold sensitive data.")
+model SecretReference {
+  @doc("The ID of an Applications.Core/SecretStore resource containing sensitive data.")
+  source: string;
+
+  @doc("The key for the secret in the secret store.")
+  key: string;
+}
+
+```        
+
+#### Usage of SecretReference in EnvironmentVariables:
+
+<a href="#secret-envvar" name="secret-envvar">:white_check_mark:</a> **DO** The above model for Secrets should be utilized for handling sensitive information through environment variables. 
+This solution is in alignment with Kubernetes design patterns. The structure also allows for environment variables to refer to other resources such as ConfigMaps, Pod Fields etc. in the future. 
+Examples include cases where environment variables containing sensitive information are injected into a container or used in a Terraform execution process.
+
+
+```
+
+@doc("environment")
+env?: Record<EnvironmentVariable>;
+
+@doc("Environment variables type")
+model EnvironmentVariable {
+
+  @doc("The value of the environment variable")
+  value?: string;
+
+  @doc("The reference to the variable")
+  valueFrom?: EnvironmentVariableReference;
+}
+
+@doc("The reference to the variable")
+model EnvironmentVariableReference {
+  @doc("The secret reference")
+  secretRef: SecretReference;
+}
+
+```
+
+#### Example in Bicep
+
+```bicep
+
+env: {
+  DB_USER: { value: 'DB_USER' }
+  DB_PASSWORD: {
+    valueFrom: {
+      secretRef: {
+        source: secret.id
+        key: 'DB_PASSWORD'
+      }
+    } 
+  }
+} 
+
+```
+
+### SecretConfig Model
+
+<a href="#secretconfig-model" name="secretconfig-model">:white_check_mark:</a> **DO** follow this structure when a component or resource in Radius requires authentication to external systems, such as private container registries, TLS certificates.
+
+```
+
+@doc("Secret Configuration used to authenticate to external systems.")
+model SecretConfig {
+  @doc("The ID of an Applications.Core/secretStores resource containing credential information.")
+  secret?: string;
+}
+
+```        
+
+#### Using SecretConfig for Git repository access
+
+<a href="#secretconfig-ext" name="secretconfig-ext">:white_check_mark:</a> **DO** In the following example SecretConfig is used to manage authentication for accessing private Terraform modules from Git repository sources.
```

Comment:
Could we generalize this further? Something like:

> :white_check_mark: **DO** use the `SecretConfig` type to reference a structured set of secret values in a resource property. Resource properties should reference a `Applications.Core/secretStores` instead of directly containing secret data.
>
> This pattern simplifies the overall design of Radius by reducing the number of places where we store secret data.